presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -236,8 +236,10 @@ public PlanNode visitGroupId(GroupIdNode node, RewriteContext<Expression> contex
         @Override
```

Comment:
```
you would need to push down it
over aggregation first, so that would mean that column A would must to
survive the aggregation which is not possible.
```

You can pushdown predicates over aggregation if predicates are on columns that are part of each grouping set